### PR TITLE
#8729 Map views plugin dev documentation

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description

Move and format documentation for dev guide

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Documentation

## Issue

**What is the current behavior?**

#8729

**What is the new behavior?**

Move and format documentation for dev guide

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] No

### CODE REVIEW

File: docs/developer-guide/maps-configuration.md [hunk 1]
```diff
@@ -135,6 +135,145 @@ Map configuration also contains the following additional options:
 - `offsetTime` the end of a time range
 - `timelineData` timeline options
 - `selectedLayer` selected layer id; if not present time cursor will be unlocked
+- `mapViews` map views options
+
+### mapViews
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:140]: I should move the whole section `## Additional map configuration options` at the end of the page.
The priority is anyway on the layers.

**Review by offtherailz — CHANGES_REQUESTED** [review]
The documentation is good. the better we have for this kind of configuration inside the map. I should only move the whole "additional" section at the end, because anyway the priority should be on map and layers.

File: docs/developer-guide/maps-configuration.md (hunk 1 above)
Comment (allyoucanmap) [reply docs/developer-guide/maps-configuration.md]: @offtherailz done

File: docs/developer-guide/maps-configuration.md [hunk 2]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+| description | string | an html string to describe the view |
+| duration | number | time duration during navigation in seconds |
+| flyTo | boolean | enable animation transition during navigation |
+| center | object | center target position as { latitude (degrees), longitude (degrees), height (meter) } |
+| cameraPosition | object | cameraPosition point of view position as { latitude (degrees), longitude (degrees), height (meter) } |
+| zoom | number | zoom level |
+| bbox | array | bounding box in WGS84 as [minx, miny, maxx, maxy] |
+| mask | object | optional configuration for the 3D tiles mask |
+| mask.enabled | boolean | enable mask |
+| mask.resourceId | string | identifier of a resource configuration |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1593]:
```suggestion
| mask.resourceId | string | identifier of a resource configuration in the `resources` array |
```

File: docs/developer-guide/maps-configuration.md [hunk 3]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+| mask.inverse | boolean | enabled inverse mask |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1594]:
```suggestion
| mask.inverse | boolean | if true enables the inverse mask |
```

File: docs/developer-guide/maps-configuration.md [hunk 4]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+| title | string | title of the view |
+| description | string | an html string to describe the view |
+| duration | number | time duration during navigation in seconds |
+| flyTo | boolean | enable animation transition during navigation |
+| center | object | center target position as { latitude (degrees), longitude (degrees), height (meter) } |
+| cameraPosition | object | cameraPosition point of view position as { latitude (degrees), longitude (degrees), height (meter) } |
+| zoom | number | zoom level |
+| bbox | array | bounding box in WGS84 as [minx, miny, maxx, maxy] |
+| mask | object | optional configuration for the 3D tiles mask |
+| mask.enabled | boolean | enable mask |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1592]:
```suggestion
| mask.enabled | boolean | enable mask |
```
```suggestion
| mask.enabled | boolean | if true enables the mask |
```

File: docs/developer-guide/maps-configuration.md [hunk 5]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+| mask.resourceId | string | identifier of a resource configuration |
+| mask.inverse | boolean | enabled inverse mask |
+| mask.offset | number | offset in meter for the inverse mask |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1595]:
```suggestion
| mask.offset | number | offset in meters for the inverse mask |
```

File: docs/developer-guide/maps-configuration.md [hunk 6]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+| terrain | object | optional configuration for terrain clipping |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1596]:
```suggestion
| terrain | object | optional configurations for terrain clipping |
```

File: docs/developer-guide/maps-configuration.md [hunk 7]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+|terrain.clippingLayerResourceId | string | identifier of a resource configuration |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1597]:
```suggestion
|terrain.clippingLayerResourceId | string | identifier of a resource configuration in the `resources` array |
```

File: docs/developer-guide/maps-configuration.md [hunk 8]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+| terrain.clippingPolygonFeatureId | string | identifier of a polygon feature available in the selected layer source |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1598]:
```suggestion
| terrain.clippingPolygonFeatureId | string | identifier of a polygonal feature available in the selected layer source to use to apply the clipping |
```

File: docs/developer-guide/maps-configuration.md [hunk 9]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+| terrain.clippingPolygonUnion | boolean | if true it applies inverse clipping |
+| globeTranslucency | object | optional configuration for the globe translucency |
+| globeTranslucency.enabled | boolean | enable translucency |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1601]:
```suggestion
| globeTranslucency.enabled | boolean | if true enables translucency |
```

File: docs/developer-guide/maps-configuration.md [hunk 10]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+| globeTranslucency.opacity | number | opacity of the globe translucency, it should be a value between 0 and 1 where 1 is fully opaque  |
+| globeTranslucency.fadeByDistance | boolean | if true the translucency is visible only in between the nearDistance and farDistance values |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1603]:
```suggestion
| globeTranslucency.fadeByDistance | boolean | if true the translucency is visible only between the `nearDistance` and `farDistance` values |
```

File: docs/developer-guide/maps-configuration.md [hunk 11]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+| globeTranslucency.nearDistance | number | minimum distance to apply translucency when fadeByDistance is true |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1604]:
```suggestion
| globeTranslucency.nearDistance | number | when `fadeByDistance` is true it indicates the minimum distance to apply translucency |
```

File: docs/developer-guide/maps-configuration.md [hunk 12]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+| globeTranslucency.farDistance | number | maximum distance to apply translucency when fadeByDistance is true |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1605]:
```suggestion
| globeTranslucency.farDistance | number |  when `fadeByDistance` is true it indicates the maximum distance to apply translucency |
```

File: docs/developer-guide/maps-configuration.md [hunk 13]
```diff
@@ -1472,3 +1459,155 @@ Due to the limitations posed by WMC format the conversion process will not prese
...
+| layers | array | array of layer configuration overrides, default property visibility and opacity |
```
Comment (offtherailz) [inline docs/developer-guide/maps-configuration.md:1606]:
```suggestion
| layers | array | array of layer configuration overrides, default properties override `visibility` and `opacity` |
```

**Review by offtherailz — CHANGES_REQUESTED** [review]
You can apply my suggestion if you agree.  
Being a part of the plugin in JSDoc is not a bad idea, but not defined as a param anyway, but in a markdown like this can be handled better also for future movements (or in a type if we were in typescript :D )

**Review by offtherailz — APPROVED** [review]

**Pull request merged**